File: smart-farm-server/src/models/Schedule.js
const mongoose = require('mongoose');

const ScheduleSchema = new mongoose.Schema(
  {
    deviceId: { type: mongoose.Schema.Types.ObjectId, ref: 'Device', required: true, index: true },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    target: { type: String, enum: ['fan', 'light', 'pump', 'main'], default: 'main' },
    action: { type: String, enum: ['ON', 'OFF'], required: true },
    time: { type: Date, required: true },
    repeat: { type: String, enum: ['daily', 'weekly'], default: 'daily' },
    active: { type: Boolean, default: true },
    lastRunAt: { type: Date },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Schedule', ScheduleSchema);

File: code_frontend/lib/models/schedule.dart
class Schedule {
  final String id;
  final String deviceId;
  final String target;
  final String action;
  final DateTime time;
  final String repeat;
  final bool active;

  Schedule({
    required this.id,
    required this.deviceId,
    required this.target,
    required this.action,
    required this.time,
    required this.repeat,
    required this.active,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['_id'],
      deviceId: json['deviceId'],
      target: json['target'] ?? 'main',
      action: json['action'],
      time: DateTime.parse(json['time']),
      repeat: json['repeat'] ?? 'daily',
      active: json['active'] ?? true,
    );
  }
}